vector-stores/spring-ai-weaviate-store/src/main/java/org/springframework/ai/vectorstore/weaviate/WeaviateVectorStore.java

```diff
@@ -548,14 +548,14 @@ private Builder(WeaviateClient weaviateClient, EmbeddingModel embeddingModel) {
 		}
 
 		/**
-		 * Configures the Weaviate object class.
-		 * @param objectClass the object class to use
+		 * Configures the Weaviate vector store option.
+		 * @param options the vector store options to use
 		 * @return this builder instance
-		 * @throws IllegalArgumentException if objectClass is null or empty
+		 * @throws IllegalArgumentException if options is null or empty
 		 */
-		public Builder objectClass(String objectClass) {
```

Comment:
We cannot remove a publicly available method in a GA release in a minor version (`1.1.x`). We should deprecate this method and encourage users to use the new API in the javadocs using the `@deprecated` tag.

Reply:
That also suggests that we may not be able to backport this PR since it adds a new API method, deprecates the existing one, etc.

Reply:
Also, since `objectClass` looks like a top-level concept in Weaviate vector store, it is also probably fine to leave it as is, and only provide the option to configure the content field. What do you think?

Reply:
@sobychacko 
Thank you for your feedback 👍

From my perspective, I don't have a strong preference either way.

The reason I did it this way is that ElasticsearchVectorStoreOptions has indexName and embeddingFieldName.
To maintain consistency, I set it up so that WeaviateVectorStoreOptions has objectClass and contentFieldName.

Reply:
I have applied all feedback except for this discussion.